### PR TITLE
Includes use of `RAILS_STRICT_WARNING` in Contribution doc - Fix #46263

### LINKED ISSUE #46263: Contributing to the Rails core should mention RAILS_STRICT_WARNINGS
Contributing to the Rails core should mention RAILS_STRICT_WARNINGS added via #46189

### Steps to reproduce
Read https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#warnings

### Expected behavior
This guide should mention that
- Rails CI will fail if the pull request raises warnings
- By setting `RAILS_STRICT_WARNINGS`, contributors can reproduce Rails CI behavior

https://github.com/rails/rails/pull/46189#event-7559954952

### Actual behavior
It does not mention that Rails CI fails if warnings are raised.

> [5.8 Warnings](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#warnings)
The test suite runs with warnings enabled. Ideally, Ruby on Rails should issue no warnings, but there may be a few, as well as some from third-party libraries. Please ignore (or fix!) them, if any, and submit patches that do not issue new warnings.

### System configuration
**Rails version**: main branch

**Ruby version**: Any version that is supported by Rails main

### PR DESCRIPTION
**Pull request by a-vzn**
### Motivation / Background

This Pull Request has been created because there should be information in the Contributing to Ruby on Rails guideline on enabling tests identical to CI during development. This information could enable contributors to iterate fewer times on broken CI tests by virtue of having the knowledge to set the `RAILS_STRICT_WARNING` flag. Enabling this flag would surface errors that need to be addressed ahead of time.

### Detail

This Pull Request changes the Contributing to Ruby on Rails markdown

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] There are no typos in commit messages and comments.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Feature branch is up-to-date with `main` (if not - rebase it).
* [x] Pull request only contains one commit for bug fixes and small features. If it's a larger feature, multiple commits are permitted but must be descriptive.
* [x] Tests are added if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.
* [x] PR is not in a draft state.
* [x] CI is passing.

### CODE REVIEW

File: guides/source/contributing_to_ruby_on_rails.md [hunk 1]
```diff
@@ -485,6 +485,7 @@ You can invoke `test_jdbcmysql`, `test_jdbcsqlite3` or `test_jdbcpostgresql` als
 ### Warnings
 
 The test suite runs with warnings enabled. Ideally, Ruby on Rails should issue no warnings, but there may be a few, as well as some from third-party libraries. Please ignore (or fix!) them, if any, and submit patches that do not issue new warnings.
+An important note is that while developing locally it is helpful to set `RAILS_STRICT_WARNINGS` to replicate Rails CI behavior. This ensures that there are no surprise failures in the CI results for a contributor's pull request.
```
Comment (yahonda) [inline guides/source/contributing_to_ruby_on_rails.md:488]: "This ensures that there are no surprise failures" is not clear what 'surprise failures' means.
Would you clarify that by setting `RAILS_STRICT_WARNINGS`, the test suite fails if warnings are raised?

**Comment by yahonda** [comment]
The update looks good. Would you squash your commits? https://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits

File: guides/source/contributing_to_ruby_on_rails.md [hunk 2]
```diff
@@ -485,6 +485,7 @@ You can invoke `test_jdbcmysql`, `test_jdbcsqlite3` or `test_jdbcpostgresql` als
 ### Warnings
 
 The test suite runs with warnings enabled. Ideally, Ruby on Rails should issue no warnings, but there may be a few, as well as some from third-party libraries. Please ignore (or fix!) them, if any, and submit patches that do not issue new warnings.
+An important note is that while developing locally it is helpful to set `RAILS_STRICT_WARNINGS` so that the testing suite will fail if any warnings are raised. This replicates Rails CI behavior and will ensure that no warnings go unnoticed during local development.
```
Comment (eileencodes) [inline guides/source/contributing_to_ruby_on_rails.md:488]:
```suggestion
Rails CI will raise if warnings are introduced. To implement the same behavior locally set `RAILS_STRICT_WARNINGS=1` when running the test suite.
```

**Review by eileencodes — CHANGES_REQUESTED** [review]
Thanks for the PR. I think the original wording is a bit of a mouthful and we can say what to do in a clearer way.

**Comment by zzak** [comment]
Combined into #47170, thanks!

**Pull request closed without merging**